tensorflow/compiler/xla/service/slice_sinker.cc

```diff
@@ -0,0 +1,273 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/xla/service/slice_sinker.h"
+#include <algorithm>
+#include <utility>
+#include <vector>
+#include "absl/types/optional.h"
+#include "absl/types/span.h"
+#include "absl/algorithm/container.h"
+#include "tensorflow/compiler/xla/shape_util.h"
+
+namespace xla {
+
+namespace {
+
+// Returns whether two slices are taken from the same indices, assuming the
+// slices are taking from tensors with the same dimensions.
+bool SameSliceConfiguration(const HloInstruction* slice_1,
+                            const HloInstruction* slice_2) {
+  CHECK(slice_1->opcode() == HloOpcode::kSlice);
```

Comment:
Let's replace CHECK(A==B) with CHECK_EQ(A, B) in these two lines. Sorry that I overlooked this even though I have read this a few times.

Reply:
done